.github/workflows/test-all.yml

```diff
@@ -31,62 +32,66 @@ env:
 
 jobs:
   build:
-    name: Build the SDK
     runs-on: ubuntu-latest
     steps:
-    # Install Chrome so the correct version of webdriver can be installed by chromedriver when
-    # setting up the repo. This must be done to build and execute Auth properly.
-    - name: install Chrome stable
-      run: |
-        npx @puppeteer/browsers install chrome@stable
-    - uses: actions/checkout@v3
-    - name: Set up Node (20)
-      uses: actions/setup-node@v3
-      with:
-        node-version: 22.10.0
-    - name: Test setup and yarn install
-      run: |
-        cp config/ci.config.json config/project.json
-        yarn
-    - name: yarn build
-      run: yarn build
-    - name: Archive build
-      if: ${{ !cancelled() }}
-      run: |
-        tar -cf build.tar --exclude=.git .
-        gzip build.tar
-    - name: Upload build archive
-      if: ${{ !cancelled() }}
-      uses: actions/upload-artifact@v3
-      with:
-          name: build.tar.gz
-          path: build.tar.gz
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version-file: '.nvmrc'
+          cache: yarn
+          cache-dependency-path: yarn.lock
+      - name: Restore cached node_modules
+        uses: actions/cache@v4
+        id: node_modules
+        with:
+          path: "**/node_modules"
+          key: node_modules-${{ runner.arch }}-${{ runner.os }}-${{ hashFiles('yarn.lock') }}
+      - run: yarn install --frozen-lockfile
+      - name: install Chrome stable
```

Comment:
The previous workflow started with the chrome installation as it seems to affect some tools installation. We're sure that it's safe to push it down this far?

Reply:
This is an oversight by me- I shouldn't have moved it down. It looks like it's passing though? https://github.com/firebase/firebase-js-sdk/actions/runs/11706621527/job/32604805660

Do you think I should move it back anyway? I'm not familiar with how this works

Reply:
I don't remember what it affected but I think it might be the chromedriver yarn install in the auth package. There might possibly be a hiccup on that install if it can't find some Chrome file. Or maybe not, maybe it doesn't matter if Chrome is there until the karma test begins to run. May be a comment somewhere in the old PRs messing with the Chrome installation in this workflow and chromedriver versions.

Reply:
I think it's worth moving it back to the top so that it stands out more.